### PR TITLE
fix 'Async function is not processed' (close #1260)

### LINKED ISSUE #1260: Async function is not processed
The used `acorn` version (4.0.1) can not parse the following code:
```js
    async hello () {}
```

### PR DESCRIPTION
**Pull request by miherlosev**
@churkin @LavrovArtem 
Changes:
- update `acorn` from 4.0.2 to 5.1.1 (latest)
- update `ACORN_UNICODE_PATCH_WARNING`, `ACORN_STRICT_MODE_PATCH_WARNING` in `test/server/script-processor-test.js`
- add test

### CODE REVIEW

**Comment by miherlosev** [comment]
@testcafe-build-bot \retest

**Comment by miherlosev** [comment]
@testcafe-build-bot \retest

**Comment by miherlosev** [comment]
@testcafe-build-bot \retest

**Comment by miherlosev** [comment]
@testcafe-build-bot \retest

**Comment by miherlosev** [comment]
@testcafe-build-bot \retest

**Comment by miherlosev** [comment]
@testcafe-build-bot \retest

**Comment by miherlosev** [comment]
FPR @churkin @AlexanderMoskovkin 
After review will be completed I will run health monitor for this branch

File: test/server/script-processor-test.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ var ACORN_UNICODE_PATCH_WARNING = multiline(function () {/*
  ```
  function readWord1() {
     ...
-    word += escStr;
+    //word += codePointToString(esc)
```
Comment (AlexanderMoskovkin) [inline test/server/script-processor-test.js]: Just to sure: is `//` should be here or it's a typo?

Comment (miherlosev) [inline test/server/script-processor-test.js]: Yes. It is type. I will fix this

**Comment by miherlosev** [comment]
@testcafe-build-bot \retest

**Comment by miherlosev** [comment]
@testcafe-build-bot  \retest

**Comment by miherlosev** [comment]
FPR

**Pull request merged**